VideoLocker/src/main/java/org/edx/mobile/user/UserAPI.java

```diff
@@ -0,0 +1,44 @@
+package org.edx.mobile.user;
+
+import android.support.annotation.NonNull;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.squareup.okhttp.OkHttpClient;
+
+import org.edx.mobile.discussion.RetroHttpExceptionHandler;
+import org.edx.mobile.http.RetroHttpException;
+import org.edx.mobile.util.Config;
+
+import retrofit.RestAdapter;
+import retrofit.client.OkClient;
+import retrofit.converter.GsonConverter;
+
+@Singleton
+public class UserAPI {
+
+    private UserService userService;
+
+    @Inject
+    public UserAPI(@NonNull Config config, @NonNull OkHttpClient client) {
+        Gson gson = new GsonBuilder()
+                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+                .setDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").create();
+
+        RestAdapter restAdapter = new RestAdapter.Builder()
+                .setClient(new OkClient(client))
+                .setEndpoint(config.getApiHostURL())
+                .setConverter(new GsonConverter(gson))
+                .setErrorHandler(new RetroHttpExceptionHandler())
+                .setLogLevel(RestAdapter.LogLevel.FULL)
+                .build();
+        userService = restAdapter.create(UserService.class);
+    }
+
+    public Account getAccount(@NonNull String username) throws RetroHttpException {
```

Comment:
> I was just copying the convention set by `DiscussionService`.

And `DiscussionService` was probably copying the convention set by `RestApiManager`, which was for compatibility purposes  :wink:

> Though I would consider `UserService` an implementation detail; it exists solely for the sake of Retrofit and so it should be hidden from the rest of the app.

If you consider Retrofit an implementation detail, then you lose one of the main benefits of using it, which is the easy interface definition it provides. Still, it's up to you I guess…